pkg/intermediate/priorityqueue.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intermediate
+
+import (
+	"container/heap"
+	"time"
+)
+
+type ItemToExpire struct {
+	// Flow related info
+	flowKey                   FlowKey
+	flowRecord                AggregationFlowRecord
+	waitForReadyToSendRetries int
+	activeExpireTime          time.Time
+	inactiveExpireTime        time.Time
+	// Index in the priority queue (heap)
+	index int
+}
+
+type TimeToExpirePriorityQueue []*ItemToExpire
+
+func (pq TimeToExpirePriorityQueue) Len() int {
+	return len(pq)
+}
+
+func (pq TimeToExpirePriorityQueue) Less(i, j int) bool {
+	if pq[i].activeExpireTime.Before(pq[i].inactiveExpireTime) {
+		if pq[j].activeExpireTime.Before(pq[j].inactiveExpireTime) {
```

Comment:
Will there be cases that `activeExpireTime` of an item is after `inactiveExpireTime`?

Reply:
There can be.. consider the default values of active expiry time to be 30s and inactive expiry time to be 45s. An inactive record first sends a flow record after 30s and resets it timer to 30s, where as inactive expiry time timer is 15s away.